packages/server/src/models/StreamSplitter.ts

```diff
@@ -117,13 +125,34 @@ export class StreamSplitter extends stream.Writable {
     this.part += 1
   }
 
+  async emitEvent<T>(name: string, payload: T) {
+    if (this.asyncEvents) {
```

Comment:
a bit less canonical but working with async events from the event emitter is also not that great 😄

Reply:
Oh! 
I just learned a neat wait to have async compatible events!

instead of calling `emit('name', payload)`

we implement our own `emitAsync` function:

```ts
  async emitEvent<T>(name: string, payload: T) {
    const listeners = this.listeners(name)
    for await (const listener of listeners) {
      await listener(payload)
    }
  }
```

and voila'